File: Week03/src/com/skillstorm/training/day4/models/Person.java
package com.skillstorm.training.day4.models;

public class Person {
	// Fully encapsulate = all the properties need to be private
	
	private String name;
	private Person bestFriend;
	
	public Person() {
		this.name = "unknown";
		
	}
	
	public Person(String name, Person bestFriend) {
		this.name = name;
		this.bestFriend = bestFriend;
	}
	
	public String getName() {
		return this.name;
	}
	
	// Setter for the name
	public void setName(String name) {
		this.name = name;
	}
	
	// Getter for bestFriend
	public Person getBestFriend() {
		return this.bestFriend;
	}
	
	// Setter for bestFriend
	public void setBestFriend(Person bestFriend) {
		this.bestFriend = bestFriend;
	}

}

File: Week15/day4/behavior-subject-example/src/app/person.ts
export class Person {

    constructor(private _name: string, private _favoriteColor: string) { }

    public get name() {
        return this._name;
    }

    public set name(name: string) {
        this._name = name;
    }

    public get favoriteColor() {
        return this._favoriteColor;
    }

    public set favoriteColor(color: string) {
        this._favoriteColor = color;
    }
}
